kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/autoconfigure/JBPMAutoConfiguration.java

```diff
@@ -388,4 +554,92 @@ public CaseInstanceMigrationService caseInstanceMigrationService(EntityManagerFa
         
         return caseInstanceMigrationService;
     }
+    
+    /*
+     * Optional quartz configuration - by default same data source is used for transactional Quartz work
+     * and new one (from properties quartz.datasource) for unmanaged access
+     */
+    
+    @Bean
+    @ConditionalOnMissingBean(name = "quartzDataSource")
+    @ConditionalOnProperty(name = {"jbpm.quartz.enabled", "jbpm.quartz.db"}, havingValue="true")
+    public DataSource quartzDataSource(DataSource dataSource) {
+        return dataSource;
+    }
+    
+    @Bean
+    @ConditionalOnMissingBean(name = "quartzDatasourceProperties")
+    @ConfigurationProperties("quartz.datasource")
+    public DataSourceProperties quartzDatasourceProperties() {
+        return new DataSourceProperties();
+    }
+    
+    @Bean
+    @ConditionalOnMissingBean(name = "quartzPoolProperties")
+    @ConfigurationProperties("quartz.datasource.dbcp2")
+    public Map<String, Object> quartzPoolProperties() {
+        return new HashMap<>();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(name = "quartzNotManagedDataSource")
+    @ConditionalOnProperty(name = {"jbpm.quartz.enabled", "jbpm.quartz.db"}, havingValue="true")
+    public DataSource quartzNotManagedDataSource() {
+        DataSource ds = quartzDatasourceProperties().initializeDataSourceBuilder().build();
+        Map<String, Object> poolProperties = quartzPoolProperties();
+        
+        MutablePropertyValues properties = new MutablePropertyValues(poolProperties);
+        new RelaxedDataBinder(ds).bind(properties);
+        
+        return ds;
+    }
+    
+    /*
+     * Helper methods
+     */
+
+    private XADataSource createXaDataSource() {
+        DataSourceProperties dataSourceProperties = dataSourceProperties();
+        
+        String className = dataSourceProperties.getXa().getDataSourceClassName();
+        if (!StringUtils.hasLength(className)) {
+            className = DatabaseDriver.fromJdbcUrl(dataSourceProperties.determineUrl())
+                    .getXaDataSourceClassName();
+        }
+        Assert.state(StringUtils.hasLength(className),
+                "No XA DataSource class name specified");
+        XADataSource dataSource = createXaDataSourceInstance(className);
+        bindXaProperties(dataSource, dataSourceProperties);
+        return dataSource;
+    }
+
+    private XADataSource createXaDataSourceInstance(String className) {
+        try {
+            Class<?> dataSourceClass = ClassUtils.forName(className, this.getClass().getClassLoader());
+            Object instance = BeanUtils.instantiate(dataSourceClass);
+            Assert.isInstanceOf(XADataSource.class, instance);
+            return (XADataSource) instance;
+        }
+        catch (Exception ex) {
+            throw new IllegalStateException(
+                    "Unable to create XADataSource instance from '" + className + "'");
```

Comment:
Maybe it would be good to include the causing exception so user knows why it couldn't be created.

Reply:
fixed, once this build is green I'll merge it